modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Composed.kt

```diff
@@ -150,6 +150,68 @@ fun <F, G> MonoidK<F>.compose(): MonoidK<Nested<F, G>> = object : ComposedMonoid
   override fun F(): MonoidK<F> = this@compose
 }
 
+interface ComposedInvariant<F, G> : Invariant<Nested<F, G>> {
+  fun F(): Invariant<F>
+
+  fun G(): Invariant<G>
+
+  override fun <A, B> Kind<Nested<F, G>, A>.imap(f: (A) -> B, g: (B) -> A): Kind<Nested<F, G>, B> =
+    F().run { unnest().imap({ ga -> G().run { ga.imap(f, g) } }, { gb -> G().run { gb.imap(g, f) } }) }.nest()
```

Comment:
There's a chance this doesn't work on Android on the D8 compiler. Can you please move each of the lambdas in imap to a separate line?